timemachine/ff/handlers/nonbonded.py

```diff
@@ -93,12 +94,19 @@ def oe_assign_charges(mol, charge_model=AM1BCCELF10):
     charge_engine = charge_engines[charge_model]
 
     oemol = convert_to_oe(mol)
-    if charge_model in (AM1ELF10, AM1BCCELF10):
+    if charge_model in ELF10_MODELS:
         oe_generate_conformations(oemol)
 
     result = oequacpac.OEAssignCharges(oemol, charge_engine)
     if result is False:
-        raise Exception("Unable to assign charges")
+        # Turn off hydrogen sampling if charge generation fails
+        # https://github.com/openforcefield/openff-toolkit/issues/346#issuecomment-505202862
+        if charge_model in ELF10_MODELS:
+            oemol = convert_to_oe(mol)
+            oe_generate_conformations(oemol, sample_hydrogens=False)
+            result = oequacpac.OEAssignCharges(oemol, charge_engine)
```

Comment:
Question: Should this branch also log a warning or raise an error under certain conditions?

Possible intent: May want know (1) whenever this branch is taken, (2) if it is taken on molecules that _don't_ contain trans-COOH, (3) if it is _not_ taken on molecules that _do_ contain trans-COOH.

Reply:
added warning